### PR TITLE
Raise nicer errors when legislation is invalid

### PR DESCRIPTION
**Pull request by michelbl**
Connected to https://github.com/openfisca/openfisca-core/issues/474

### CODE REVIEW

File: tests/test_legislations.py [hunk 1]
```diff
@@ -13,54 +13,8 @@
 
 
 def check_legislation_xml_file(year):
-    legislation_tree = conv.check(legislationsxml.make_xml_legislation_info_list_to_xml_element(False))(
-        tax_benefit_system.legislation_xml_info_list, state = conv.default_state)
-    legislation_xml_json = conv.check(legislationsxml.xml_legislation_to_json)(
-        legislation_tree,
-        state = conv.default_state,
-        )
+    compact_legislation = tax_benefit_system.get_legislation(year)
```
Comment (fpagnoux) [inline tests/test_legislations.py]: What are we testing exactly ? @michelbl do you think it is relevant ?
I don't know much this part of OpenFisca.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+### 18.2.2 - [#742](https://github.com/openfisca/openfisca-france/pull/742)
```
Comment (fpagnoux) [inline CHANGELOG.md]: As you are not modifying the `openfisca_france` package, you don't need to publish a new version.

File: tests/test_legislations.py (hunk 1 above)
Comment (michelbl) [inline tests/test_legislations.py]: We are testing that the xml files can load. I think it is relevant, even if a lot of other tests load them.

File: CHANGELOG.md (hunk 2 above)
Comment (michelbl) [inline CHANGELOG.md]: Only the tests are changed. But CI will complain if I don't bump the version number ?

Comment (fpagnoux) [inline CHANGELOG.md]: The CI should not enforce a version bump if you haven't change any file inside `openfisca_france ` ;)

File: tests/test_legislations.py (hunk 1 above)
Comment (fpagnoux) [inline tests/test_legislations.py]: 👌

File: CHANGELOG.md (hunk 2 above)
Comment (fpagnoux) [inline CHANGELOG.md]: However, on a second thoughts, improving the error is a proper nice feature, so bumping the version number is legit.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
+* Amélioration technique
+* Détails :
+  - Met à jour les tests sur la législation à la version `Openfisca-Core` `12.1.1`.
```
Comment (fpagnoux) [inline CHANGELOG.md]: Du point de vue de l'utilisateur, ce qui change n'est pas l'adaptation des tests, mais l'amélioration du message d'erreur en cas d'erreur dans la législation.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,9 +2,9 @@
 
 ### 18.2.2 - [#742](https://github.com/openfisca/openfisca-france/pull/742)
 
-* Amélioration technique
+* Amélioration de messages d'erreur
```
Comment (fpagnoux) [inline CHANGELOG.md]: Presque : cette première ligne est normalisée : il n'y a que [4 valeurs possibles](https://github.com/openfisca/openfisca-france/blob/master/CONTRIBUTING.md#1315---684)

**Pull request merged**